homeassistant/components/ecobee/climate.py

```diff
@@ -653,7 +679,11 @@ def set_temperature(self, **kwargs):
 
     def set_humidity(self, humidity):
         """Set the humidity level."""
+        if humidity not in range(0, 101):
+            raise ValueError("Invalid set_humidity value (must be in range 0-100): %s" % humidity)
```

Comment:
Please use an f-string instead of old style string formatting.

Reply:
Okay, I changed that and added another condition so the humidity controls will only show up if a humidifier is installed and set in "manual" mode.